pkg/package-server/provider/registry.go

```diff
@@ -189,6 +189,15 @@ func (p *RegistryProvider) syncCatalogSource(obj interface{}) (syncError error)
 		Namespace: source.GetNamespace(),
 		Name:      source.GetName(),
 	}
+
+	logger.Infof("updating PackageManifest based on CatalogSource changes: %v", key)
+	timeout, cancel := context.WithTimeout(context.Background(), cacheTimeout)
+	defer cancel()
+	client, err := p.registryClient(key)
+	if err == nil {
+		err = p.refreshCache(timeout, client)
+	}
```

Comment:
I see that your adding this code to check if the list of packages has changed. The code looks correct in that it will update the list of packagemanifests but I have two questions:
1. Could you make sure that the error is captured when the PackageServer is unable to refresh the cache? Either as a log or returning the error?
2. It looks like this could be moved into the [if statement below](https://github.com/operator-framework/operator-lifecycle-manager/blob/006c12bdbeda179ed0b24193e639596dfd4569ef/pkg/package-server/provider/registry.go#L201) to prevent it from being called when sources don't already exist. Could you investigate?

Reply:
1. wow that was a very nice catch, I assumed the return was err instead of syncError. Kinda a weird way to express dislike for this kind of error return method.
2. good call, that would reduce the traffic. I somehow thought about it backward initially.